cpp/oneapi/dal/backend/interop/table_conversion.hpp

```diff
@@ -122,12 +123,159 @@ inline daal::data_management::NumericTablePtr convert_to_daal_table(const homoge
     return copy_to_daal_homogen_table<Data>(table);
 }
 
+template <typename T>
+inline auto convert_to_daal_csr_table(array<T>& data,
+                                      array<std::int64_t>& column_indices,
+                                      array<std::int64_t>& row_indices,
+                                      std::int64_t row_count,
+                                      std::int64_t column_count,
+                                      bool allow_copy = false) {
+    if (!data.get_count() || !column_indices.get_count() || !row_indices.get_count())
+        return daal::services::SharedPtr<daal::data_management::CSRNumericTable>();
+
+    if (allow_copy) {
+        data.need_mutable_data();
+        column_indices.need_mutable_data();
+        row_indices.need_mutable_data();
+    }
+
+    ONEDAL_ASSERT(data.has_mutable_data());
+    ONEDAL_ASSERT(column_indices.has_mutable_data());
+    ONEDAL_ASSERT(row_indices.has_mutable_data());
+
+    const auto daal_data =
+        daal::services::SharedPtr<T>(data.get_mutable_data(), daal_object_owner{ data });
+    ONEDAL_ASSERT(sizeof(std::size_t) == sizeof(std::int64_t));
+    const auto daal_column_indices = daal::services::SharedPtr<std::size_t>(
+        reinterpret_cast<std::size_t*>(column_indices.get_mutable_data()),
```

Comment:
Will it work on MacOS where `std::uint64_t` and `std::size_t` are different types?

Reply:
I've checked reinterpret_cast from size_t* to int64_t* (similar, from int64_t* to size_t*), it works correctly